ios/dtx_codec/channel.go

```diff
@@ -49,19 +49,31 @@ func (d *Channel) ReceiveMethodCall(selector string) Message {
 // MethodCall is the standard DTX style remote method invocation pattern. The ObjectiveC Selector goes as a NSKeyedArchiver.archived NSString into the
 // DTXMessage payload, and the arguments are separately NSKeyArchiver.archived and put into the Auxiliary DTXPrimitiveDictionary. It returns the response message and an error.
 func (d *Channel) MethodCall(selector string, args ...interface{}) (Message, error) {
-	payload, _ := nskeyedarchiver.ArchiveBin(selector)
 	auxiliary := NewPrimitiveDictionary()
 	for _, arg := range args {
 		auxiliary.AddNsKeyedArchivedObject(arg)
```

Comment:
I'm curious if it would be better if we checked the type of `arg` here and decide if we wanna use `AddNsKeyedArchivedObject()` vs `AddInt32()`? That way the callee wouldn't have to worry about creating a new primitive dictionary and the need for the `MethodCallWithAuxiliary` would disappear.

Reply:
for instance, this is the log, if for `killPid` we just add the int32:
```
{"channel_id":"com.apple.instruments.server.services.processcontrol","error":"Timed out waiting for response for message:2 channel:1","level":"info","methodselector":"killPid:","msg":"failed starting invoking method","time":"2024-11-19T09:24:50-06:00"}
```